e2e-tests/development-runtime/plugins/gatsby-source-fake-data/gatsby-node.js

```diff
@@ -29,6 +29,9 @@ exports.sourceNodes = async function sourceNodes({
     reporter.info(`Webhook data detected; creating nodes`)
     webhookBody.items.forEach(node => createNode(api.getNode(node, helpers)))
   } else {
+    if (!(webhookBody && webhookBody[`fake-data-update`])) {
+      await api.reset()
+    }
```

Comment:
This is meant to use "initial" state, unless we want to trigger refresh/update explicitly for this plugin